src/collision/collider/mod.rs

```diff
@@ -42,16 +45,167 @@ pub trait IntoCollider<C: AnyCollider> {
+    #[cfg_attr(
+        feature = "2d",
+        doc = "# use avian2d::{prelude::*, math::{Vector, Scalar}};"
+    )]
+    #[cfg_attr(
+        feature = "3d",
+        doc = "# use avian3d::{prelude::*, math::{Vector, Scalar}};"
+    )]
+    /// # use bevy::prelude::*;
+    /// # use bevy::ecs::system::{SystemParam, lifetimeless::{SRes, SQuery}};
+    ///
+    /// #[derive(Component)]
+    /// pub struct VoxelData {
+    ///     // collider voxel data...
+    /// }
+    ///
+    /// #[derive(Component)]
+    /// pub struct VoxelCollider;
+    ///
+    /// # impl ComputeMassProperties2d for VoxelCollider {
+    /// #     fn mass(&self, density: f32) -> f32 {0.}
+    /// #     fn unit_angular_inertia(&self) -> f32 { 0.}
+    /// #     fn center_of_mass(&self) -> Vec2 { Vec2::ZERO }
+    /// # }
+    ///
+    /// # impl ComputeMassProperties3d for VoxelCollider {
+    /// #     fn mass(&self, density: f32) -> f32 {0.}
+    /// #     fn unit_principal_angular_inertia(&self) -> Vec3 { Vec3::ZERO }
+    /// #     fn center_of_mass(&self) -> Vec3 { Vec3::ZERO }
+    /// # }
+    ///
+    /// impl AnyCollider for VoxelCollider {
+    ///     type Context = (
+    ///         // you can query data here
+    ///         SQuery<&'static VoxelData>,
+    ///         // or put any other read-only system param here
+    ///         SRes<Time>,
+    ///     );
+    ///
+    /// #   fn aabb_with_context(
+    /// #       &self,
+    /// #       _: AabbContext<Self::Context>,
+    /// #       _: Vector,
+    /// #       _: impl Into<Rotation>,
+    /// #   ) -> ColliderAabb { unimplemented!() }
+    ///
+    ///     fn contact_manifolds_with_context(
+    ///         &self,
+    ///         other: &Self,
+    ///         context: ContactManifoldContext<Self::Context>,
+    ///         position1: Vector,
+    ///         rotation1: impl Into<Rotation>,
+    ///         position2: Vector,
+    ///         rotation2: impl Into<Rotation>,
+    ///         prediction_distance: Scalar,
+    ///     ) -> Vec<ContactManifold> {
+    ///         let [voxels1, voxels2] = context.0.get_many([context.entity1, context.entity2])
+    ///             .expect("our own `VoxelCollider` entities should have `VoxelData`");
+    ///         let elapsed = context.1.elapsed();
+    ///         // do some computation...
+    /// #       unimplemented!()
+    ///     }
+    /// }
+    /// ```
+    type Context: for<'w, 's> ReadOnlySystemParam<Item<'w, 's>: Send + Sync>;
+
     /// Computes the [Axis-Aligned Bounding Box](ColliderAabb) of the collider
     /// with the given position and rotation.
     #[cfg_attr(
         feature = "2d",
         doc = "\n\nThe rotation is counterclockwise and in radians."
     )]
-    fn aabb(&self, position: Vector, rotation: impl Into<Rotation>) -> ColliderAabb;
+    fn aabb_with_context(
+        &self,
+        context: AabbContext<Self::Context>,
+        position: Vector,
+        rotation: impl Into<Rotation>,
+    ) -> ColliderAabb;
```

Comment:
Personally this argument order still feels a bit weird to me. The parameters of the AABB are really the main thing here, and the context is just an extra thing to provide more contextual data. I think it'd be more natural to have the context be the last argument here and in the other `_with_context` methods.